pkg/trigger/controller.go

```diff
@@ -47,25 +50,40 @@ const (
 )
 
 type MigrationTrigger struct {
-	client            migrationclient.Interface
-	migrationInformer cache.SharedIndexInformer
-	queue             workqueue.RateLimitingInterface
+	client                 migrationclient.Interface
+	kubeClient             kubernetes.Interface
+	migrationInformer      cache.SharedIndexInformer
+	storageVersionInformer cache.SharedIndexInformer
+	queue                  workqueue.RateLimitingInterface
+	storageVersionQueue    workqueue.RateLimitingInterface
 	// The timestamp of last time discovery is performed.
 	heartbeat metav1.Time
+
+	lastSeenTransitionTime map[string]metav1.Time
 }
 
-func NewMigrationTrigger(c migrationclient.Interface) *MigrationTrigger {
+func NewMigrationTrigger(c migrationclient.Interface, kubeClient kubernetes.Interface) *MigrationTrigger {
 	mt := &MigrationTrigger{
-		client: c,
+		client:     c,
+		kubeClient: kubeClient,
 		// TODO: share one with the kubemigrator.go.
 		migrationInformer: controller.NewStatusAndResourceIndexedInformer(c),
-		queue:             workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "migration_triggering_controller"),
+		storageVersionInformer: apiserverinternalinformers.NewStorageVersionInformer(kubeClient,
+			0,
+			cache.Indexers{}),
+		queue:                  workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "migration_triggering_controller"),
```

Comment:
Can we name this queue and respective enqueue/dequeue function as well? Maybe just `migrationQueue`.